tdp/core/deployment/deployment_plan.py

```diff
@@ -192,60 +192,47 @@ def from_failed_deployment(dag: Dag, deployment_log: DeploymentLog):
-            DeploymentTypeEnum.DAG,
-            DeploymentTypeEnum.RECONFIGURE,
-        ):
-            original_operations = DeploymentPlan.from_dag(
-                dag,
-                deployment_log.targets,
-                deployment_log.sources,
-                deployment_log.filter_expression,
-                deployment_log.filter_type,
-                deployment_log.restart,
-            ).operations
-            original_operation_names = [
-                operation.name for operation in original_operations
-            ]
-        elif deployment_log.deployment_type in (
-            DeploymentTypeEnum.OPERATIONS,
-            DeploymentTypeEnum.RESUME,
-        ):
-            original_operation_names = deployment_log.targets
-            original_operations = [
-                dag.collections.operations[name] for name in original_operation_names
-            ]
-        else:
+        if not isinstance(deployment_log.deployment_type, DeploymentTypeEnum):
             raise UnsupportedDeploymentTypeError(
-                f"Resuming from a {deployment_log.deployment_type} is not supported"
+                f"Resuming from a {deployment_log.deployment_type} is not supported."
             )
 
+        # TODO: fix the query to retrieve the operations in the right order
+        previous_operations = sorted(
+            deployment_log.operations, key=lambda operation: operation.operation_order
+        )
+
         if len(deployment_log.operations) > 0:
-            last_operation_log = deployment_log.operations[-1]
-            try:
-                index = original_operation_names.index(last_operation_log.operation)
-            except ValueError as e:
-                raise GeneratedDeploymentPlanMissesOperationError(
-                    f"'{last_operation_log}' is not in the reconstructed operation list from database parameters",
-                    original_operation_names,
-                ) from e
-            remaining_operations = original_operations[index:]
-            remaining_operation_names = original_operation_names[index:]
-        else:
-            remaining_operations = original_operations
-            remaining_operation_names = original_operation_names
+            failed_operation_id = next(
+                (
+                    i
+                    for i, operation in enumerate(previous_operations)
+                    if operation.state == OperationStateEnum.FAILURE
+                ),
+                None,
+            )
+            operations_names_to_resume = [
+                operation.operation
+                for operation in previous_operations[failed_operation_id:]
+            ]
+            operations_to_resume = [
+                Operation(operation_name)
+                for operation_name in operations_names_to_resume
+            ]
 
         new_deployment_log = DeploymentLog(
-            targets=remaining_operation_names,
+            targets=operations_names_to_resume,
```

Comment:
Why do we put all the held operations to targets? If it is not, absence of targets doesn't change anything, no?

Reply:
For logging purpose, this allow use to know what is resuming directly from the deployment list (I did not change this behavior).

I might suggest some improvements for the database schema later, but that's not a priority as it works for now.